components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/impl/MLModelHandler.java

```diff
@@ -437,13 +435,14 @@ public MLModel retrieveModel(long modelId) throws MLModelHandlerException {
      * @param modelId Unique ID of the built ML model.
      * @throws MLModelPublisherException
      */
-    public void publishModel(int tenantId, String userName, long modelId) throws MLModelPublisherException {
+    public void publishModel(int tenantId, String userName, long modelId) throws InvalidRequestException, MLModelPublisherException {
         InputStream in = null;
+        String errorMsg = "Failed to publish the model [id] ";
```

Comment:
You know the model id here it self right? so why not add it to the message here.